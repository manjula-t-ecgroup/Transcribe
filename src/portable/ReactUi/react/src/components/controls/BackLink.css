.BackLink {
    color: rgba(202, 208, 212, 0.5);
    font-size: 28pt;
    text-align: center;
    width: 100px;
}

.BackLink *::before {
    color: rgba(202, 208, 212, 0.5);
}

.BackLink a {
    text-decoration: none;
    color: rgba(202, 208, 212, 0.5);
}