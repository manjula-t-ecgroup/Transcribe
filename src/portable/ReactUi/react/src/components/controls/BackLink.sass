@import "../../_variables.sass"

.BackLink 
	color: $label
	font-size: 28pt
	text-align: center
	width: 100px
	*::before 
		color: rgba(202, 208, 212, 0.5)
	
	a 
		text-decoration: none
		color: rgba(202, 208, 212, 0.5)
		&:hover 
			-moz-border-radius: 5px
			border-radius: 5px
			-moz-box-shadow: 0px 0px 30px 0px $white
			box-shadow: 0px 0px 30px 0px $white