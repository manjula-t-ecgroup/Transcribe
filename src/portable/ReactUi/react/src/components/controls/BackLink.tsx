import * as React from 'react';
import { Link } from 'react-router-dom';
import './BackLink.sass';

interface IProps {
    target: string;
};

class BackLink extends React.Component<IProps, object> {
    public render() {
        const { target } = this.props;
        return (
            <div id="BackLink" className="BackLink">
                <Link to={target}>
                    {"\u276E"}
                </Link>
            </div>
        )
    }
};

export default BackLink;
