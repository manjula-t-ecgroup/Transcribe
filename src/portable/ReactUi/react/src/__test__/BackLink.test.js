import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer'
import BackLink from '../components/controls/BackLink';

configure({ adapter: new Adapter() })

// Snapshot for BackLink
describe('>>>Control: Backlink --- Snapshot',()=>{
    it('+++capturing Snapshot of BackLink', () => {
        const renderedValue =  renderer.create(<Router><BackLink target={"/root"}/></Router>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: BackLink', () => {
    let wrapper;
    const minProps = {
        target: "/root",
    };

    beforeEach(()=>{
        wrapper = shallow(<BackLink {...minProps} />)
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
