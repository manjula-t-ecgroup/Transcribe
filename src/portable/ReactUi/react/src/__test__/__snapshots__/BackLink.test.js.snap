// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: Backlink --- Snapshot +++capturing Snapshot of BackLink 1`] = `
<div
  className="BackLink"
  id="BackLink"
>
  <a
    href="/root"
    onClick={[Function]}
  >
    ‚ùÆ
  </a>
</div>
`;
